extensions/amp-font/0.1/amp-font.js

```diff
@@ -0,0 +1,316 @@
+   * @private
+   */
+  forceLoadFont_() {
+    var  hasWidthChanged = false;
+    var hasHeightChanged = false;
+    var ncustomFontFamily_ = normalizeFontFamily_(this.customFontFamily_);
+    DEFAULT_FONTS_.some(font => {
+      if(this.hasTimedOut_) {
+        this.onFontLoadError_();
+        return true;
+      }
+      let normalizedFont = normalizeFontFamily_(font);
+      let defaultFontElementId =
+          '-amp-font-' + ncustomFontFamily_ + '-' + normalizedFont;
+      let element = this.document_.getElementById(defaultFontElementId);
+      let currentFontFamily = element.style.fontFamily;
+      if (currentFontFamily.indexOf(this.customFontFamily_) == -1) {
+        element.style.fontFamily =
+            this.customFontFamily_ + ',' + currentFontFamily;
+      }
+      var defaultDimensions =
+          this.measurements_[defaultFontElementId];
+       hasWidthChanged = (
+        Math.abs(
+          defaultDimensions.width - element./*OK*/offsetWidth) > TOLERANCE_);
+      hasHeightChanged = (
+        Math.abs(
+          defaultDimensions.height - element./*OK*/offsetHeight) > TOLERANCE_);
+      if ( hasWidthChanged || hasHeightChanged) {
+        return true;
+      }
+    });
+    if ( hasWidthChanged || hasHeightChanged) {
+      this.onFontLoadSuccess_();
+    } else if(this.hasTimedOut_) {
+      this.onFontLoadError_();
+    } else {
+      RAF_(() => this.forceLoadFont_());
+    }
+  }
+
+
+  /**
+   * @private
+   */
+  onFontLoadSuccess_() {
+    var addClassName = this.element.getAttribute('on-load-class') || '';
+    this.documentElement_.classList.add(addClassName);
+    this.dispose_();
+  }
+
+
+  /**
+   * @private
+   */
+  onFontLoadError_() {
+    var addClassName = this.element.getAttribute('on-error-class') || '';
+    this.documentElement_.classList.add(addClassName);
+    this.dispose_();
+  }
+
+
+  /**
+   * @private
+   */
+  dispose_() {
+    var removeClassName =
+        this.element.getAttribute('while-loading-class') || '';
+    this.documentElement_.classList.remove(removeClassName);
+    this.measurements_ = Object.create(null);
+    let containerElement = this.document_.getElementById(
+       '-amp-font-container-'+normalizeFontFamily_(this.customFontFamily_));
+    if (containerElement) {
+      containerElement.parentNode.removeChild(containerElement);
+    }
+    if (this.timeoutHandle_) {
+      clearTimeout(this.timeoutHandle_);
+    }
+  }
+
+  /**
+   * Starts the timeout counter to not attempt loading the font after the
+   * timeout  normal.
+   * @private
+   */
+  beginTimeout_() {
+    var timeoutAttribute = this.element.getAttribute('timeout');
+    var timeoutInMs;
+    timeoutInMs = (timeoutAttribute.indexOf('s') > -1) ?
+        parseInt(timeoutAttribute, 10) * 1000 : parseInt(timeoutAttribute, 10);
+    timeoutInMs = timeoutInMs ? timeoutInMs : DEFAULT_TIMEOUT_;
+
+    this.timeoutHandle_ = timer.delay(() => {
+      this.hasTimedOut_ = true;
```

Comment:
I think it's worth to check font loaded one last time. The reason is because we can't always control animation frames and sometimes we don't even know that we can't control them. This would be a good fallback for those cases.